### Задание 1

Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов.

Даны значения роста в трех группах случайно выбранных спортсменов:
Футболисты: 173, 175, 180, 178, 177, 185, 183, 182
Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180
Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170

_Проведем однофакторный дисперсионный анализ для определения различий в среднем росте среди групп футболистов, хоккеистов и штангистов. Это позволит нам определить, есть ли статистически значимые различия между средними значениями роста в этих трех группах._

- Нулевая гипотеза (H0​): Средний рост в трех группах одинаков.
- Альтернативная гипотеза (H1​): Средний рост в трех группах отличается.

In [10]:
import numpy as np
import scipy.stats as stats

# Данные роста
football_players = [173, 175, 180, 178, 177, 185, 183, 182]
hockey_players = [177, 179, 180, 188, 177, 172, 171, 184, 180]
weightlifters = [172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]

# Проведение однофакторного дисперсионного анализа
f_stat, p_value = f_oneway(football_players, hockey_players, weightlifters)

print(f"F-статистика: {f_stat}, p-value: {p_value}")

F-статистика: 5.500053450812596, p-value = 0.010482206918698694


Результаты однофакторного дисперсионного анализа:

Поскольку p-value (0.0105) меньше уровня значимости (0.05), мы отвергаем нулевую гипотезу.
Это означает, что есть статистически значимые различия в среднем росте среди групп футболистов, хоккеистов и штангистов.

In [13]:
# Проведение теста Крускала-Уоллиса
h_stat, p_value = stats.kruskal(football_players, hockey_players, weightlifters)

print(f"H-статистика: {h_stat}, p-value: {p_value}")

H-статистика: 7.897493213863828, p-value: 0.01927885061595347


Результаты теста Крускала-Уоллиса:

Поскольку p-value (0.0193) меньше уровня значимости (0.05), мы отвергаем нулевую гипотезу.
Это означает, что есть статистически значимые различия в среднем росте среди групп футболистов, хоккеистов и штангистов.

- Как и в случае с однофакторным дисперсионным анализом, тест Крускала-Уоллиса показал наличие статистически значимых различий среди групп.
- Однофакторный дисперсионный анализ проверяет различия в средних значениях, тогда как тест Крускала-Уоллиса проверяет различия в распределениях.

Оба теста дают схожий вывод: средний рост среди трех групп спортсменов различается.